### ISSUE TITLE
Assert failure in CREATE STATISTICS

### ISSUE THREAD

**Issue by marcocitus** [citusdata/citus#4518]
I seem to get a crash / assert failure in CREATE STATISTICS when wrapping the column names in parentheses:
```sql
create table test (x int, y int);
select create_distributed_table('test','x');

create statistics stats_xy on (x, y) from test;
server closed the connection unexpectedly
        This probably means the server terminated abnormally
        before or while processing the request.
The connection to the server was lost. Attempting reset: Succeeded.
```

Backtrace:
```
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007f6c550f0859 in __GI_abort () at abort.c:79
#2  0x00005610b5b7fd71 in ExceptionalCondition (conditionName=conditionName@entry=0x7f6c52ee2e90 "!(((((const Node*)(column))->type) == T_ColumnRef))", 
    errorType=errorType@entry=0x7f6c52ed7553 "FailedAssertion", fileName=fileName@entry=0x7f6c52ee2e68 "deparser/deparse_statistics_stmts.c", lineNumber=lineNumber@entry=241)
    at assert.c:54
#3  0x00007f6c52e4bc29 in AppendColumnNames (buf=buf@entry=0x7fff777a2db0, stmt=stmt@entry=0x5610b6e46df0) at deparser/deparse_statistics_stmts.c:241
#4  0x00007f6c52e4be2d in AppendCreateStatisticsStmt (buf=buf@entry=0x7fff777a2db0, stmt=stmt@entry=0x5610b6e46df0) at deparser/deparse_statistics_stmts.c:137
#5  0x00007f6c52e4bef4 in DeparseCreateStatisticsStmt (node=0x5610b6e46df0) at deparser/deparse_statistics_stmts.c:44
#6  0x00007f6c52e49294 in DeparseTreeNode (stmt=stmt@entry=0x5610b6e46df0) at deparser/deparse.c:36
#7  0x00007f6c52e38daa in PreprocessCreateStatisticsStmt (node=0x5610b6e46df0, queryString=<optimized out>) at commands/statistics.c:84
#8  0x00007f6c52e40739 in multi_ProcessUtility (pstmt=0x5610b6e46d58, queryString=0x5610b6dabb78 "create statistics foos on (x, y) from test;", 
    context=PROCESS_UTILITY_TOPLEVEL, params=0x0, queryEnv=0x0, dest=0x5610b6dace88, completionTag=0x7fff777a3440 "") at commands/utility_hook.c:387
#9  0x00007f6c52dd7d71 in CallInnerUtilityHook (pstmt=<optimized out>, queryString=queryString@entry=0x5610b6dabb78 "create statistics foos on (x, y) from test;", 
    context=<optimized out>, params=<optimized out>, queryEnv=<optimized out>, dest=<optimized out>, completionTag=0x7fff777a3440 "") at src/pg_migrator_source.c:180
#10 0x00007f6c52dd80c7 in PgMigratorSourceProcessUtility (pstmt=<optimized out>, queryString=0x5610b6dabb78 "create statistics foos on (x, y) from test;", 
    context=<optimized out>, params=<optimized out>, queryEnv=<optimized out>, dest=<optimized out>, completionTag=0x7fff777a3440 "") at src/pg_migrator_source.c:130
#11 0x00005610b5a55cff in ProcessUtility (pstmt=pstmt@entry=0x5610b6dacd90, queryString=<optimized out>, context=<optimized out>, params=<optimized out>, 
    queryEnv=<optimized out>, dest=dest@entry=0x5610b6dace88, completionTag=0x7fff777a3440 "") at utility.c:356
#12 0x00005610b5a51fd3 in PortalRunUtility (portal=portal@entry=0x5610b6e86288, pstmt=pstmt@entry=0x5610b6dacd90, isTopLevel=isTopLevel@entry=true, 
    setHoldSnapshot=setHoldSnapshot@entry=false, dest=dest@entry=0x5610b6dace88, completionTag=completionTag@entry=0x7fff777a3440 "") at pquery.c:1175
#13 0x00005610b5a52c40 in PortalRunMulti (portal=portal@entry=0x5610b6e86288, isTopLevel=isTopLevel@entry=true, setHoldSnapshot=setHoldSnapshot@entry=false, 
    dest=dest@entry=0x5610b6dace88, altdest=altdest@entry=0x5610b6dace88, completionTag=completionTag@entry=0x7fff777a3440 "") at pquery.c:1321
#14 0x00005610b5a538f0 in PortalRun (portal=portal@entry=0x5610b6e86288, count=count@entry=9223372036854775807, isTopLevel=isTopLevel@entry=true, run_once=run_once@entry=true, 
    dest=dest@entry=0x5610b6dace88, altdest=altdest@entry=0x5610b6dace88, completionTag=0x7fff777a3440 "") at pquery.c:796
#15 0x00005610b5a4fb97 in exec_simple_query (query_string=query_string@entry=0x5610b6dabb78 "create statistics foos on (x, y) from test;") at postgres.c:1215
#16 0x00005610b5a51b1b in PostgresMain (argc=<optimized out>, argv=argv@entry=0x5610b6e4a2f0, dbname=<optimized out>, username=<optimized out>) at postgres.c:4247
#17 0x00005610b59c098e in BackendRun (port=port@entry=0x5610b6e43d00) at postmaster.c:4494
#18 0x00005610b59c3e88 in BackendStartup (port=port@entry=0x5610b6e43d00) at postmaster.c:4177
#19 0x00005610b59c40de in ServerLoop () at postmaster.c:1725
#20 0x00005610b59c5671 in PostmasterMain (argc=3, argv=<optimized out>) at postmaster.c:1398
#21 0x00005610b591ed1d in main (argc=3, argv=0x5610b6da4b80) at main.c:228
```

**Comment by marcocitus** [comment]
Also happens for expressions:
```sql
create statistics stats_xy on x+1 from test;
```

**Comment by marcocitus** [comment]
We're supposed to get an error:
```sql
create statistics stats_xy on (x,y) from local;
ERROR:  only simple column references are allowed in CREATE STATISTICS
```

**Comment by marcocitus** [comment]
fixed by #4523
